### PR TITLE
UI and set parking spaces implemented

### PR DESCRIPTION
**Pull request by SofiaZarzabal**
- First UI implement
- Set parking spaces functionality implement
- Unit testing for each methods in presenter

**Folder Structure**
image

**Demostration in Pixel 3a**
GIF PIXEL 3A KOTLIN

**Demostration in Nexus S**
GIF NEXUS S KOTLIN

**Unit Test**
image

**Coverage**
image

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ConfigureParkingDialogView.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ConfigureParkingDialogContract.ConfigureParkingDialogView {
+
+    override fun closeDialog() {
+        fragment?.let {
+            val dialog: DialogFragment = it as DialogFragment
+            dialog.dismiss()
+        }
+    }
+
+    override fun getParkingSpaces(): String {
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ConfigureParkingDialogView.kt]: you can change this by 

`    override fun getParkingSpaces(): String = binding.inputFragmentDialog.text.toString()`

to be more kotlin style :D

File: app/src/main/java/com/example/parkingsystemkotlin/utils/ExtensionFunctions.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.example.parkingsystemkotlin.utils
+
+import android.content.Context
+import android.widget.Toast
+
+fun Context.toast(message: String) =
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemkotlin/utils/ExtensionFunctions.kt:6]: this is beautiful ❤️

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -4,15 +4,29 @@
...
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
+    <Button
+        android:id="@+id/button_main_select_parking"
+        style="@style/ButtonMainActivitySelectParkingSpacesStyle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/margin_top_main_activity_button"
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: I think that you can add this to the style

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -4,15 +4,29 @@
...
     tools:context=".MainActivity">
 
     <TextView
+        android:id="@+id/title_main_activity"
+        style="@style/TitleMainActivityStyle"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Hello World!"
-        app:layout_constraintBottom_toBottomOf="parent"
+        android:layout_marginTop="@dimen/margin_top_main_activity_text"
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: I think that you can add this to the style

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystemkotlin/utils/ExtensionFunctions.kt (hunk 2 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystemkotlin/utils/ExtensionFunctions.kt:6]: Thank you so much! <3

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ConfigureParkingDialogView.kt (hunk 1 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ConfigureParkingDialogView.kt]: Done!

File: app/src/main/res/layout/activity_main.xml (hunk 4 above)
Comment (SofiaZarzabal) [reply app/src/main/res/layout/activity_main.xml]: Done!

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (SofiaZarzabal) [reply app/src/main/res/layout/activity_main.xml]: Done!

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**